### PR TITLE
Deck editor

### PR DESCRIPTION
**Pull request by BenMatteson**
Did a bunch of stuff to improve the deck editor, mostly hotkey behavior.

### CODE REVIEW

File: octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml.cs [hunk 1]
```diff
@@ -645,21 +645,28 @@ private void AddResultCard(object sender, SearchCardIdEventArgs e)
...
+            Expander presenter = (Expander)VisualTreeHelper.GetChild(cont, 0);
+            if (presenter.IsExpanded)
+            {
+                // Bring change into view
+                DataGrid grid = (DataGrid)presenter.Content;
+                DataGridRow row = (DataGridRow)grid.ItemContainerGenerator.ContainerFromItem(element);
+                if (row == null) // new card, added at bottom
+                {
+                    // would be nice to get rid of the magic numbers, but woks unless we ever change the size of the grid
+                    grid.BringIntoView(new Rect(0, grid.ActualHeight - 30, grid.ActualWidth, 60)); // ~last two rows after new card added
```
Comment (kellyelton) [inline octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml.cs]: Why doesn't something like this work?
`grid.FirstDisplayedScrollingRowIndex = grid.RowCount-1;`

File: octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml [hunk 2]
```diff
@@ -39,6 +39,7 @@
                             <TextBlock.FontWeight>
                                 <MultiBinding Converter="{StaticResource ActiveSectionConverter}">
                                     <Binding Path="ActiveSection" ElementName="self" />
+                                    <Binding />
                                 </MultiBinding>
                             </TextBlock.FontWeight>
                         </TextBlock>
```
Comment (kellyelton) [inline octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml]: Couldn't you also do this?


```
<TextBlock.FontWeight>
    <Binding Path="ActiveSection" ElementName="self" Converter="{StaticResource ActiveSectionConverter}"/>
</TextBlock.FontWeight>
```

File: octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml.cs (hunk 1 above)
Comment (BenMatteson) [reply octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml.cs]: The problem is that the row doesn't exist yet, the card was added to the source, but the grid hasn't updated, so it doesn't have a row to bring into view. 
I tried handling the bring into view elsewhere but I couldn't figure out where I could do it...

File: octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml (hunk 2 above)
Comment (BenMatteson) [reply octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml]: probably if I rewrote the converter. This is just how it used to work before I deleted the second `<binding />` ages ago, breaking it. I Honestly have no idea what that one is even passing, I just dug deep into the blames to figure out what it was supposed to look like.

File: octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml.cs (hunk 1 above)
Comment (BenMatteson) [reply octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml.cs]: oh, also, it's a DataGrid, not a DataGridView, so that doesn't exits.

Comment (kellyelton) [reply octgnFX/Octgn/DeckBuilder/DeckBuilderWindow.xaml.cs]: Oh gotcha.

**Pull request merged**